scaling_performance/host_practices.adoc

```diff
@@ -136,6 +136,62 @@ endif::[]
 etcd processes are typically memory intensive. Master / API server processes are CPU intensive. This makes them a reasonable co-location pair within a single machine or virtual machine (VM). Optimize communication between etcd and master hosts either by co-locating them on the same host, or providing a dedicated network.
 ====
 
+[[providing-storage-to-an-etcd-node-using-pci-passthrough]]
+=== Providing Storage to an etcd Node Using PCI Passthrough
+
+To provide fast storage to an etcd node so that etcd is stable at large scale,
+use PCI passthrough. To set this up with Red Hat OpenStack 11 or later:
+
+. Ensure Intel Vt-x is enabled in BIOS.
+
+. Enable the input–output memory management unit (IOMMU). In the
+*_/etc/sysconfig/grub_* file, add `intel_iommu=on iommu=pt` to the end of the
+`GRUB_CMDLINX_LINUX` line, within the quotation marks.
+
+. Regenerate *_/etc/grub2.cfg_* by running:
++
+----
+$ grub2-mkconfig -o /etc/grub2.cfg
+----
+
+. Reboot the system.
+
+. On controllers in *_/etc/nova.conf_*:
++
+----
+[filter_scheduler]
+
+enabled_filters=RetryFilter,AvailabilityZoneFilter,RamFilter,DiskFilter,ComputeFilter,ComputeCapabilitiesFilter,ImagePropertiesFilter,ServerGroupAntiAffinityFilter,ServerGroupAffinityFilter,PciPassthroughFilter
+
+available_filters=nova.scheduler.filters.all_filters
+
+[pci]
+
+alias = { "vendor_id":"144d", "product_id":"a820",
+"device_type":"type-PCI", "name":"nvme" }
+----
+
+. On compute nodes in *_/etc/nova/nova.conf_*:
++
+----
+[pci]
+
+passthrough_whitelist = { "address": "0000:06:00.0" }
+
+alias = { "vendor_id":"144d", "product_id":"a820",
+"device_type":"type-PCI", "name":"nvme" }
+----
++
+To retrieve the required `address`, `vendor_id`, and `product_id` values, run:
++
+----
+# lspci -nn | grep devicename
+----
+
```

Comment:
Although the OpenStack docs contain more details as stated later, let us just mention that the user needs to configure and OpenStack flavor to use the NVMe and launch the etcd node using this flavor.